src/driver/compile.ml

```diff
@@ -222,16 +231,66 @@ let link : compiled list -> _ =
         (match c.m with
         | Module _ -> Atomic.incr Stats.stats.linked_units
         | Mld _ -> Atomic.incr Stats.stats.linked_mlds);
-        { output_file = Fpath.(set_ext "odocl" c.output_file); src = None }
+        {
+          output_file = Fpath.(set_ext "odocl" c.output_file);
+          src = None;
+          pkgname = c.pkgname;
+        }
         :: impl
   in
   Fiber.List.map link compiled |> List.concat
 
+let odoc_index_path ~odoc_dir pkgname =
+  Fpath.(odoc_dir / pkgname / "index.odoc-index")
```

Comment:
I believe it would be cleaner to output index files outside of the directory made for `.odocl` files. That is:
```suggestion
  Fpath.(index_dir / pkgname / "index.odoc-index")
```
or 
```suggestion
  Fpath.(index_dir / pkgname ^ ".odoc-index")
```

Although, this is not mandatory. Just resolve the comment if you believe it is better like that.